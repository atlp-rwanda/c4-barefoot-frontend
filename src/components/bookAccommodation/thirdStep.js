import React,{useEffect,useState} from "react";
import {Card,CardContent, CardActionArea, CardActions, CardMedia,Typography,Grid,Divider,Button,FormControlLabel,Checkbox,TextField} from '@material-ui/core';
import { Field, Form, Formik,  FormikConfig,FormikValues} from 'formik';
import {getAccommodationsByLocation,selectAccommodation} from "../../redux/actions/fetchAccommodationByLocation";
import {getAccommodation,getAccommodations,getAccommodationAminity} from "../../redux/actions/fetchAccommodations";
import {bookAccommodations,clearBookSnackbar} from "../../redux/actions/bookAccommodationAction";
import {getTemperature} from "../../redux/actions/getWeather";
import { connect } from 'react-redux';
import AccommodationCard from "../AccommodationCardWithReview";
import { makeStyles } from '@material-ui/core/styles';
import { Label, Place } from '@material-ui/icons'
import colors from '../colors';
import { Pagination } from '@material-ui/lab';
import Ratings from '../RatingStars';
import { Skeleton } from '@material-ui/lab';
import * as yup from 'yup';
import CloudIcon from '@material-ui/icons/Cloud';
import Loader from '../Loader'
import {DatePicker} from '@material-ui/pickers';

const useStyles = makeStyles((theme) => ({
    // root: {
    //   maxWidth: 345,
    //   height: 360,
    //   height: '100%',
    //   display: 'flex',
    //   flexDirection: 'column',
    //   justifyContent: 'space-between'
    // },
    media: {
      height: 440
    },
    checkbox:{
        display: 'block',
        position:'absolute',
        color: 'secondary',
        right: 0,
        fontSize:'25px'
    },
    cardActions:{
        display: 'flex',
        flexDirection:'column',
        alignItems:'flex-start',
    },
    separate:{
        // marginBottom:theme.spacing(3),
        marginLeft:theme.spacing(3)
    },
    divider:{
        // marginBottom:theme.spacing(3),
        marginTop:theme.spacing(4)
    },
    separator:{
        marginBottom:theme.spacing(3),
        marginTop:theme.spacing(3),
        marginLeft:theme.spacing(2),
    },
    dates:{
        marginRight:theme.spacing(3),
        marginLeft:theme.spacing(2)
    },
    reviews:{
        fontSize: '14px',
        color: colors.primary100,
    },
    cardContent:{
      overflow: 'hidden'
    },
    container:{
        marginLeft:theme.spacing(9),
        // width:'80%'
    },
    btn:{
        display:'flex',
        alignContent:'flex-end',
        flexDirection:'column',
        alignItems:'flex-start',
    },
    btncontainer1:{
        display:"flex",
        justifyContent:"flex-start"
        // width:'80%'
    },
    btncontainer2:{
        display:"flex",
        justifyContent:"flex-end"
        // width:'80%'
    },
    titleText:{
      [theme.breakpoints.down('sm')]:{
        fontSize:'18px',
      }
    },
    item:{
        display:'block'
    }
  }));
function Home(props){
    
  const classes = useStyles();
    let temp= null;
    const [Fromdate,setFromDate] = useState(null)
    const [Returndate,setRetrunDate] = useState(null)
    if(props.temp){
        temp=<Typography gutterBottom variant="h6"  className={classes.titleText} >
                <CloudIcon color="primary"/> {props.temp-273.15}                                  
            </Typography>
    }

    const handleBook=(value)=>{
        let formdata={From:"",To:""}
        formdata.From=value.From
        formdata.To=value.To
        props.bookAccommodations(props.selectedAccommodation,formdata,props.nextStep)
    }
    const validationSchema = yup.object().shape({
        From: yup
        .date()
        .required('Booking Date is required',"Booking Date is required")
        .min(new Date(),"You can not book a Hotel in the past"),
        To: yup
            .date()
            .required('Checkout Date is required',"Che")
            .min(yup.ref('From'),"You can not checkout before you arrive  ")
    })

    return(
        <React.Fragment>
            <Loader open={props.status}/>
            <Card>
                <CardContent>
                    <Formik initialValues={{
                        From:null,
                        To:null
                    }}
                    validationSchema={validationSchema}
                    onSubmit={(values, {setSubmitting, resetForm}) => {
                            handleBook(values)
                            resetForm()
                        }}
                    >
                    {({ values,errors,touched,handleChange,handleBlur,handleSubmit,isSubmitting}) => (
                        <Form onSubmit={handleSubmit}>
                            <div className={classes.divider} >
                            <Card className={classes.root} >
                                {(!props.accommodation ? 
                                    <CardActionArea>
                                    <Skeleton animation="wave" variant="rect" className={classes.media} />
                                    <CardContent className={classes.cardContent} >
                                        <Skeleton animation="wave" height={30} width="60%" />
                                        <Skeleton animation="wave" height={10} width="80%" />
                                    </CardContent>
                                    <CardActions className={classes.cardActions}>
                                        <Skeleton animation="wave" height={30} width="60%" />
                                    </CardActions>
                                    </CardActionArea>
                                    :
                                <>
                                <CardActionArea>
                                    
                                    <CardMedia
                                    // onClick={handleViewMore}
                                    className={classes.media}
                                    image={props.accommodation.photos}
                                    title={props.accommodation.title}
                                    />
                                </CardActionArea>
                                    <CardContent  className={classes.cardContent} >
                                        <Typography gutterBottom variant="h5" component="h2" className={classes.titleText}>
                                        {props.accommodation.title}
                                        </Typography>
                                        <Typography variant="body2" color="textSecondary" component="p" noWrap >
                                            {props.accommodation.description}
                                        </Typography>
                                        <div className={classes.divider}>
                                            <Typography variant="h4" color="primary">
                                                Booking
                                            </Typography>
                                            <Grid container spacing={3} className={classes.divider} >
                                                <Grid container item xs={3} spacing={3} className={classes.dates}>
                                                    {/* <Field 
                                                         name='fromDate'
                                                        label='Book From *'
                                                        margin='normal'
                                                        as={<DatePicker value={Fromdate} onChange={setFromDate}
                                                            autoOk
                                                            inputVariant="outlined"
                                                            variant="inline"
                                                            emptyLabel="Book From"
                                                            openTo='year'
                                                            format="MMMM dd yyyy"
                                                            disablePast
                                                            />}
                                                        error={touched.fromDate && errors.fromDate}
                                                        helperText={touched.fromDate && errors.fromDate}
                                                                                                                                                /> */}
                                                                                                                                                <TextField 
                                                            id="From"
                                                            label="Book From"
                                                            type="datetime-local"
                                                            error={touched.From && errors.From}
                                                            helperText={touched.From && errors.From}
                                                            InputLabelProps={{
                                                                shrink: true
                                                            }}
                                                            format="yyy-dd-mm HH:MM:ss"
                                                            onChange={handleChange}
                                                            onBlur={handleBlur}
                                                            value={values.From}
                                                        />
                                                        {/* <Error touched={touched.fromDate} message={errors.fromDate}/> */}
                                                        {/* <Error touched={touched.fromDate} /> */}
                                                </Grid>
                                                <Grid container item xs={3} spacing={3} className={classes.dates}>
                                                    {/* <DatePicker value={Returndate} onChange={setRetrunDate}
                                                        autoOk
                                                        inputVariant="outlined"
                                                        variant="inline"
                                                        emptyLabel="Book To"
                                                        openTo='year'
                                                        format="MMMM dd yyyy"
                                                        disablePast
                                                        /> */}
                                                        <TextField 
                                                            id="To"
                                                            label="Book upto"
                                                            type="datetime-local"
                                                            error={touched.To && errors.To}
                                                            helperText={touched.To && errors.To}
                                                            InputLabelProps={{
                                                                shrink: true
                                                            }}
                                                            format="yyy-dd-mm HH:MM:ss"
                                                            onChange={handleChange}
                                                            onBlur={handleBlur}
                                                            value={values.To}
                                                        />
                                                        {/* <Error touched={touched.returnDate} message={errors.returnDate}/> */}
                                                </Grid>
                                            </Grid>
                                        </div>
                                    </CardContent>
                                </>
                                )}
                                </Card>
                            </div>
                            <div className={classes.textCenter,classes.divider}>
                            <div className={classes.textCenter,classes.divider,classes.btncontainer1}>
                                <Button
                                    color='primary'
                                    variant='contained'
                                    id="backBtn"
                                    className={classes.button}
                                    onClick={() => props.prevStep()}
                                >
                                    Back
                                </Button>
                            </div>
                            <div className={classes.textCenter,classes.divider,classes.btncontainer2}>
                                <Button
                                    type='submit'
                                    color='primary'
                                    variant='contained'
                                    btn='submitBtn'
                                    disabled={isSubmitting}
                                    className={classes.button}
                                >
                                    Book
                                </Button>
                            </div>
                            </div>
                        </Form>
                        )}
                    </Formik>
                </CardContent>
            </Card>
        </React.Fragment>
    )
}


const mapStateToProps=state=>({
    accommodations:state.fetchAccommodations.accommodationsByLocation,
    accommodation:state.fetchAccommodations.accommodation,
    selectedAccommodation:state.fetchAccommodations.selectedAccommodation,
    count:state.fetchAccommodations.count,
    status:state.bookAccommodations.pending,
    amenities:state.fetchAccommodations.amenities,
    temp:state.fetchAccommodations.temp
})
export default connect(mapStateToProps,{getAccommodationsByLocation,selectAccommodation,getAccommodation,getAccommodations,getAccommodationAminity,getTemperature,bookAccommodations,clearBookSnackbar}) (Home)