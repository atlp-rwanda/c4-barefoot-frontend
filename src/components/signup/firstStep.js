import React from 'react';
import PropTypes from 'prop-types';
import { Formik, Form, Field } from 'formik';
import {TextField, InputAdornment, Box, Avatar} from '@material-ui/core';
import { AccountCircle, Email, Lock } from "@material-ui/icons";
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import * as yup from 'yup';
import { Link, BrowserRouter as Router, Route } from 'react-router-dom';
import Skeleton from '@material-ui/lab/Skeleton';
import { Login } from '../views/Login';

const useStyles = makeStyles(theme => ({
  form: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    '@media(max-width: 320px)' : {
      display: 'unset'
    },
  },
  button: {
    margin: theme.spacing(1)
  },
  stepButtons: {
    display: 'grid',
    gridTemplateColumns: ' auto 1fr auto',
    alignItems: 'center',
    '@media(max-width: 740px)' : {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-around'
    },
  },
  box:{
    display: 'flex',
    alignItems: 'center',
    height: '48px',
    marginTop: '16px',
    marginBottom: '8px',
  },
  textBox: {
    height: '38px',
    marginLeft: '0.5rem',
    width: '100%',
  },
  skeletonButton: {
    width: '68px',
    height: '38px',
    borderRadius: '5px'
  },
  skeletonLink: {
    width: '120px',
    borderRadius: '5px'
  }
}));

const validationSchema = yup.object({
    first_name: yup
        .string()
        .required('First Name is required')
        .max(20),
    last_name: yup
        .string()
        .required('Last Name is required')
        .max(20),
    username: yup
        .string()
        .required('Username is required')
        .max(20),
    email: yup
        .string()
        .email('Invalid email')
        .required('Email is required'),
    password: yup
        .string()
        .required("Password is required")
        .min(8),
    confirmPassword: yup
        .string()
        .required("Confirm Password")
        .when("password", {
        is: val => (val && val.length > 0 ? true : false),
        then: yup.string().oneOf(
            [yup.ref("password")],
            "Passwords must be equel"
        )
    })

});

export const FirstStep = ({ loading, formData, setFormData, nextStep }) => {
  const classes = useStyles();
  return (
    <>
      <Formik
        initialValues={formData}
        onSubmit={values => {
          setFormData(values);
          nextStep();
        }}
        validationSchema={validationSchema}
      >
        {({ errors, touched }) => (
          <Form className={classes.form} form-data='form-1'>
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
              
            ) : (
              <Field
                name='first_name'
                label='First Name *'
                margin='normal'
                as={TextField}
                error={touched.first_name && errors.first_name}
                helperText={touched.first_name && errors.first_name}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <AccountCircle />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            {loading ? (
              <Box className={classes.box}>
              <Skeleton variant="circle">
                <Avatar/>
              </Skeleton>
              <Skeleton variant="rect" className={classes.textBox} />
            </Box>
            ) : (
              <Field
                name='last_name'
                label='Last Name *'
                margin='normal'
                as={TextField}
                error={touched.last_name && errors.last_name}
                helperText={touched.last_name && errors.last_name}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <AccountCircle />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                name='username'
                label='Username *'
                margin='normal'
                as={TextField}
                error={touched.username && errors.username}
                helperText={touched.username && errors.username}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <AccountCircle />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                type='email'
                name='email'
                label='Email *'
                margin='normal'
                as={TextField}
                error={touched.email && errors.email}
                helperText={touched.email && errors.email}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Email />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                type='password'
                name='password'
                label='Password *'
                margin='normal'
                as={TextField}
                error={touched.password && errors.password}
                helperText={touched.password && errors.password}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Lock />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            {loading ? (
              <Box className={classes.box}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.textBox} />
              </Box>
            ) : (
              <Field
                type='password'
                name='confirmPassword'
                label='Confirm Password *'
                margin='normal'
                as={TextField}
                error={touched.confirmPassword && errors.confirmPassword}
                helperText={touched.confirmPassword && errors.confirmPassword}
                InputProps={{
                  startAdornment: (
                    <InputAdornment position="start">
                      <Lock />
                    </InputAdornment>
                  ),
                }}
              />
            )}
            
            <div className={classes.stepButtons}>
              {loading ? (
                <Box className={classes.box}>
                  <Skeleton variant="rect" className={classes.skeletonLink} />
                </Box>
              ) : (
                <a href='/login' style={{textDecoration:'none'}}>Sign in instead</a>
              )}
              
              <div></div>
              { loading ? (
              <Box className={classes.box}>
                <Skeleton variant="rect" className={classes.skeletonButton} />
              </Box>
              
            ) : (
              <Button
                type='submit'
                variant='contained'
                color='primary'
                className={classes.button}
              >
                Next
              </Button>
            )}
              
            </div>
          </Form>
        )}
      </Formik>
    </>
  );
};

FirstStep.propTypes = {
  formData: PropTypes.object.isRequired,
  setFormData: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired
};
