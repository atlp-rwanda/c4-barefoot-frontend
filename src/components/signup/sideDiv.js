import React from 'react'
import { makeStyles } from '@material-ui/core/styles';
import Skeleton from '@material-ui/lab/Skeleton';
import { Box } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
    sideNav: {
        display: 'flex',
        flexDirection: 'column',
        flexWrap: 'nowrap',
        justifyContent: 'center',
        alignContent: 'center',
        fontSize: '1.5rem',
        '@media(max-width: 740px)' : {
            width: '80%'
        }
    },
    locationImg: {
        width: '70%',
        '@media(max-width: 740px)' : {
            display: 'none',
        }
    },
    img:{
        textAlign: 'center'
    },
    imgSkeleton: {
        width: '70%',
        maxWidth: '300px',
        minHeight: '248px',
        marginLeft: 'auto',
        marginRight: 'auto',
        '@media(max-width: 740px)' : {
            display: 'none'
        }
    },
    sideText:{
        textAlign: 'center',
        color: 'dodgerblue',
        '@media(max-width: 740px)' : {
            fontSize: '18px',
            marginBottom: '20px'
          },
          '@media(max-width: 360px)' : {
            marginBottom: '0'
          }
    },
    textBox: {
        height: '25px',
        marginLeft: 'auto',
        marginRight: 'auto',
        marginBottom: '0.5rem',
        width: '100%',
        maxWidth: '300px'
      },
  }));
function SideDiv({loading}){
    const classes = useStyles();
    return(
        <div className={classes.sideNav}>
            { loading ? (
                <Box className={classes.box}>
                    <Skeleton variant="rect" className={classes.textBox}/>
                    <Skeleton variant="rect" className={classes.textBox} />
                </Box>
            ) : (
                <div className={classes.sideText}>
                    <p>Please create account to get </p>
                    <p>started with Barefoot Nomard.</p>
                </div>
            )
            }
            {loading ? (
                <Box className={classes.box}>
                    <Skeleton variant="rect" className={classes.imgSkeleton}/>
                </Box>
            ) : (
                <div className={classes.img}>
                    <img className={classes.locationImg} src={'https://res.cloudinary.com/barefoot-nomad-app/image/upload/v1609407601/images/location_goruvq.png'} alt='Barefoot Nomards'/>
                </div>
            )}
        </div>
    )
}

export default SideDiv