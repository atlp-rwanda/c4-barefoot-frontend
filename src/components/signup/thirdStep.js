import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux'
import { makeStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import { List, ListItem, ListItemText } from '@material-ui/core/';
import { AccountCircle, Email, BusinessCenter, Language, Subject, EventBusy } from "@material-ui/icons";
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Avatar from '@material-ui/core/Avatar';
import Snackbar from '@material-ui/core/Snackbar';
import MuiAlert from '@material-ui/lab/Alert';
import Loader from '../Loader'
import { requestSignup } from '../../redux/actions/signupRequestAction'
import { useDispatch, useSelector } from 'react-redux'
import { closeSnackbar } from '../../redux/actions/signupRequestAction';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles(theme => ({
  container: {
    textAlign: 'left'
  },
  textCenter: {
    textAlign: 'center'
  },
  button: {
    margin: theme.spacing(1)
  },
  large: {
    width: theme.spacing(10),
    height: theme.spacing(10),
  }
}));


export const Confirm = ({ formData, prevStep, nextStep }) => {
  const { t, i18n } = useTranslation();
  const classes = useStyles();
  const { first_name, last_name, email, username, occupation, bio, address, language, profile_picture} = formData;

  const dispatch = useDispatch();
  const requestingState = useSelector(state => state.signup.requesting);
  const openError = useSelector(state => state.signup.errorOpen);
  const errorMsg = useSelector(state => state.signup.error);

  const signupRequest = () => {
    const{confirmPassword, ...user} = {...formData}
    dispatch(requestSignup(user, nextStep))
  }
  const handleClose = () => {
    dispatch(closeSnackbar());
  };
  function Alert(props) {
    return <MuiAlert elevation={6} variant="filled" {...props}  itemID='alert' />;
  }
  return (
    <>
      <Loader open={requestingState} />
      <Snackbar open={openError} autoHideDuration={6000} onClose={handleClose} >
        <Alert onClose={handleClose} severity="error" >
          Error: {errorMsg ? JSON.stringify(errorMsg).replace(/[\\'"]+/g, '') : 'Error Not set'}
        </Alert>
      </Snackbar>
      <div>
        <List>
        <ListItem>
            <Avatar alt="user" name='file' src={profile_picture ? profile_picture : ''} className={classes.large}/>
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary={t('Names')}
              secondary={first_name + ' ' + last_name}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary={t('Username')}
              secondary={username}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Email/>
            </ListItemIcon>
            <ListItemText
              primary={t('Email')}
              secondary={email}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <AccountCircle />
            </ListItemIcon>
            <ListItemText
              primary={t('Occupation')}
              secondary={occupation}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <BusinessCenter />
            </ListItemIcon>
            <ListItemText
              primary={t('Address')}
              secondary={address}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Language />
            </ListItemIcon>
            <ListItemText
              primary={t('Language')}
              secondary={language}
              
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <Subject />
            </ListItemIcon>
            <ListItemText
              primary={t('Biography')}
              secondary={bio}
            />
          </ListItem>
          <ListItem>
            <ListItemIcon>
              <EventBusy />
            </ListItemIcon>
            <ListItemText
              primary={t('Occupation')}
              secondary={occupation}
            />
          </ListItem>
        </List>
        <div className={classes.textCenter}>
          <Button
            color='secondary'
            variant='contained'
            id="backBtn"
            className={classes.button}
            onClick={() => prevStep()}
          >
            {t("Back")}
          </Button>

          <Button
            color='primary'
            variant='contained'
            btn='submitBtn'
            className={classes.button}
            onClick={() => signupRequest()}
          >
            {t("Submit")}
          </Button>
        </div>
      </div>
    </>
  );
};

Confirm.propTypes = {
  formData: PropTypes.object.isRequired,
  prevStep: PropTypes.func.isRequired,
  nextStep: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  requestSignup: state.signup,
})

export default connect(mapStateToProps, { requestSignup })(Confirm)