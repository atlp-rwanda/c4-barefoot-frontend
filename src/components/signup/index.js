import { FirstStep } from './firstStep';
import { SecondStep } from './secondStep';
import { Confirm } from './thirdStep';
import SideNav from './sideDiv'
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';
import StepLabel from '@material-ui/core/StepLabel';
import Typography from '@material-ui/core/Typography';
import SocialButtons from '../signup/socialButton'
import { Avatar, Box } from '@material-ui/core';
import Skeleton from '@material-ui/lab/Skeleton';
import { useTranslation } from 'react-i18next';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    textAlign: 'center',
    float: 'right',
    backgroundColor: '#fff',
  },
  rightSide: {
    border: '2px solid lightblue',
    padding: '1.5rem',
    '@media(max-width: 740px)' : {
      maxWidth: '450px',
    },
    '@media(max-width: 370px)' : {
      border: 'none',
      padding: 'unset'
    },
  },
  h3:{
    color: 'dodgerblue',
    margin: '1em',
    '@media(max-width: 740px)' : {
      fontSize: '15px'
    }
  },
  container: {
    display: 'grid',
    fontFamily: "'Poppins', sans-serif",
    '@media(max-width: 740px)' : {
      display: 'flex',
      flexDirection: 'column'
    },
    gridTemplateColumns: '1fr 1fr', 
    alignItems: 'center',
    justifyContent: 'center',
    padding: '1rem',
    boxSizing: 'border-box',
    marginBottom: '100px'
  },
  backButton: {
    marginRight: theme.spacing(1),
  },
  instructions: {
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(1),
  },
  textBox: {
    height: '20px',
    marginLeft: 'auto',
    marginRight: 'auto',
    marginBottom: '0.5rem',
    width: '100%',
    maxWidth: '200px'
  },
  box: {
    display: 'grid',
    alignItems: 'center'
  },
  stepperContainer: {
    display: 'flex',
    justifyContent: 'center',
    width: '400px',
    marginTop: '4px',
    
    '@media(max-width: 800px)' : {
      width: '330px'
    },
    '@media(max-width: 740px)' : {
      width: 'unset'
    },
  },
  stepperBox: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    minWidth: '100px'
  },
  stepperText: {
    height: '10px',
    width: '70%',
    marginTop: '5px'
  }
}));

function getSteps() {
  return ['Basic Information', 'More information', 'All information'];
}

function getStepContent(loading, stepIndex, formData, setFormData, handleNext, handleBack) {
  switch (stepIndex) {
    case 0:
      return (
        <FirstStep
          formData={formData}
          setFormData={setFormData}
          nextStep={handleNext}
          loading = {loading}
        />
      );
    case 1:
      return (
        <SecondStep
          formData={formData}
          setFormData={setFormData}
          nextStep={handleNext}
          prevStep={handleBack}
          loading = {loading}
        />
      );
    case 2:
      return (
        <Confirm formData={formData} nextStep={handleNext} prevStep={handleBack} loading = {loading} />
      );
    default:
  }
}

export default function SignUp() {
  const { t, i18n } = useTranslation();
  const [formData, setFormData] = React.useState({
    first_name: '',
    last_name: '',
    email: '',
    occupation: '',
    username: '',
    bio: '',
    address: '',
    language: '',
    password: '',
    profile_picture: ''
  });
  const classes = useStyles();
  const [activeStep, setActiveStep] = React.useState(0);
  const [loading, setLoading] = React.useState(true)
  const steps = getSteps();

  const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);
  };

  const handleBack = () => {
    setActiveStep((prevActiveStep) => prevActiveStep - 1);
  };

  setTimeout(() => { setLoading(false)}, 1000);
  return (
    <div className={classes.container}>
      <SideNav loading={loading}/>
      <div className={classes.root}>
        <div className={classes.rightSide}>
          { loading ? (
            <Box className={classes.box}>
              <Skeleton variant="rect" className={classes.textBox}/>
            </Box>
          ) : (
            <h3 className={classes.h3}>{t("Signup with your credentials")}</h3>
          )}
          
          { loading ? (
            <Box className={classes.stepperContainer}>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
              <Box className={classes.stepperBox}>
                <Skeleton variant="circle">
                  <Avatar/>
                </Skeleton>
                <Skeleton variant="rect" className={classes.stepperText}/>
                <Skeleton variant="rect" className={classes.stepperText}/>
              </Box>
            </Box>
          ) : (
            <Stepper activeStep={activeStep} alternativeLabel>
              {steps.map((label) => (
                <Step key={label}>
                  <StepLabel>{label}</StepLabel>
                </Step>
              ))}
            </Stepper>
          )}
          
          <div>
            {activeStep === steps.length ? (
              <div>
                <Typography className={classes.instructions}>Well Done <i class="fa fa-thumbs-up"></i></Typography>
                <Typography className={classes.instructions}>Now, Check your inbox <p>to verify your account</p></Typography>
              </div>
            ) : (
              <div>
                <Typography className={classes.instructions}>{getStepContent(loading, activeStep, formData, setFormData, handleNext, handleBack)}</Typography>
              </div>
            )}
          </div>
        </div>
        <SocialButtons stepIndex={activeStep} loading={loading}/>
      </div>
    </div>
  );
}

