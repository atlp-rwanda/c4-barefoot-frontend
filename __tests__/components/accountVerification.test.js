import { shallow, mount } from "enzyme";
import React from 'react';
import toJson from "enzyme-to-json";
import { Confirm } from "../../src/components/signup/thirdStep";
import VerifyAccount from "../../src/components/signup/verifyAccount";


describe('Account verifacation', () => {
    let formData = new FormData()

    const props = {
        formData:formData,
        setFormData: function(){},
    }
    describe('Account verification Rendering', ()=>{
        
        it('renders Account varidation correctly', () => {
            const wrapper = mount(<VerifyAccount />)
            expect(toJson(wrapper)).toMatchSnapshot();
        });
        
    })

})
