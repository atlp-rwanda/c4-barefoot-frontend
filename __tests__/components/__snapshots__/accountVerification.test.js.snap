// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account verifacation Account verification Rendering renders Account varidation correctly 1`] = `
<verifyAccount>
  <div
    className="makeStyles-container-1"
  >
    <Styled(MuiBox)
      className="makeStyles-boxContainer-2"
    >
      <div
        className="MuiBox-root MuiBox-root-6 makeStyles-boxContainer-2"
      >
        <Styled(MuiBox)>
          <div
            className="MuiBox-root MuiBox-root-7"
          >
            <WithStyles(ForwardRef(Skeleton))
              height={40}
              variant="text"
              width={300}
            >
              <ForwardRef(Skeleton)
                classes={
                  Object {
                    "circle": "MuiSkeleton-circle",
                    "fitContent": "MuiSkeleton-fitContent",
                    "heightAuto": "MuiSkeleton-heightAuto",
                    "pulse": "MuiSkeleton-pulse",
                    "rect": "MuiSkeleton-rect",
                    "root": "MuiSkeleton-root",
                    "text": "MuiSkeleton-text",
                    "wave": "MuiSkeleton-wave",
                    "withChildren": "MuiSkeleton-withChildren",
                  }
                }
                height={40}
                variant="text"
                width={300}
              >
                <span
                  className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style={
                    Object {
                      "height": 40,
                      "width": 300,
                    }
                  }
                />
              </ForwardRef(Skeleton)>
            </WithStyles(ForwardRef(Skeleton))>
            <WithStyles(ForwardRef(Skeleton))
              height={40}
              variant="text"
              width={300}
            >
              <ForwardRef(Skeleton)
                classes={
                  Object {
                    "circle": "MuiSkeleton-circle",
                    "fitContent": "MuiSkeleton-fitContent",
                    "heightAuto": "MuiSkeleton-heightAuto",
                    "pulse": "MuiSkeleton-pulse",
                    "rect": "MuiSkeleton-rect",
                    "root": "MuiSkeleton-root",
                    "text": "MuiSkeleton-text",
                    "wave": "MuiSkeleton-wave",
                    "withChildren": "MuiSkeleton-withChildren",
                  }
                }
                height={40}
                variant="text"
                width={300}
              >
                <span
                  className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
                  style={
                    Object {
                      "height": 40,
                      "width": 300,
                    }
                  }
                />
              </ForwardRef(Skeleton)>
            </WithStyles(ForwardRef(Skeleton))>
          </div>
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Skeleton))
          height={50}
          style={
            Object {
              "marginLeft": "auto",
              "marginRight": "auto",
            }
          }
          variant="rect"
          width={70}
        >
          <ForwardRef(Skeleton)
            classes={
              Object {
                "circle": "MuiSkeleton-circle",
                "fitContent": "MuiSkeleton-fitContent",
                "heightAuto": "MuiSkeleton-heightAuto",
                "pulse": "MuiSkeleton-pulse",
                "rect": "MuiSkeleton-rect",
                "root": "MuiSkeleton-root",
                "text": "MuiSkeleton-text",
                "wave": "MuiSkeleton-wave",
                "withChildren": "MuiSkeleton-withChildren",
              }
            }
            height={50}
            style={
              Object {
                "marginLeft": "auto",
                "marginRight": "auto",
              }
            }
            variant="rect"
            width={70}
          >
            <span
              className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style={
                Object {
                  "height": 50,
                  "marginLeft": "auto",
                  "marginRight": "auto",
                  "width": 70,
                }
              }
            />
          </ForwardRef(Skeleton)>
        </WithStyles(ForwardRef(Skeleton))>
      </div>
    </Styled(MuiBox)>
  </div>
</verifyAccount>
`;
